### DISCUSSION TITLE
Modifying config.env in setupNodeEvents

### DISCUSSION THREAD

**Discussion by redbugz** [currents-dev/cypress-currents-reporter#108] [Q&A] [answered]
I noticed it's recommended to use this instead of cy2 now. I haven't had the specific problems with cy2 and Cypress 12.8.x yet that it warns about yet, but I thought I would give it a try to prepare for what seems the inevitable future? But I ran into an issue trying cypress-cloud. We're a bit new to sorry-cypress.

When I ran this command:
```
npx cypress-cloud --parallel --record --ci-build-id=hello-cy-cloud-1
```
I got this error:
The first problem was this error:
```
Error: Can't run because no spec files were found.

We searched for specs matching this glob pattern:

  > /Users/user/code/app-dir/dwigyshyei
....
ERROR  Unable to resolve cypress configuration
- make sure that 'cypress-cloud/plugin' is installed
- report the issue together with cypress stdout and stderr
```
After some triage, I determined it was because we are modifying config.env in the `setupNodeEvents`

```
    setupNodeEvents(on, config) {
/// a bunch of code to do some things based on env vars, including modifying config.env
      config.env = {
        ...config.env,
        ...envs, // some custom envs that we use for some tasks
      }

      return cloudPlugin(on, config)
    }
```

If I comment out the lines that set config.env, then it finds our specs and proceeds past that error (I get a different error I made an issue about), It doesn't matter what I do to config.env, the actual keys/values don't seem to matter, just any modification of any kind. So I assume there must be some kind of hashing going on that links that generated spec glob pattern to the config.env?
When we use cy2 it works fine modifying config.env

So my question is it wrong to be modifying the config.env in that function, or should cypress-cloud handle that situation, or is there a better way?

**Comment by agoldis** [comment, marked as answer]
`cloudPlugin(on, config)` indeed [expects](https://github.com/currents-dev/cypress-cloud/blob/main/packages/cypress-cloud/plugin/index.ts#L18) some `env` variables from `config` passed to `setupNodeEvents(on, config)`.

If you're modifying the config, please make sure to preserve the original env values.

As an alternative, please lift `cloudPlugin` invocation (currently it doesn't return any value)

```
  setupNodeEvents(on, config) {
     cloudPlugin(on, config)
     /// a bunch of code to do some things based on env vars, including modifying config.env
    }
```

**Reply by redbugz** [reply]
I did confirm the values past in were preserved, I will try calling cloudPlugin first

**Reply by redbugz** [reply]
Running cloudPlugin first worked
